packages/ec-linear/src/genome/bitstring.rs

```diff
@@ -129,28 +131,46 @@ impl Linear for Bitstring {
     }
 }
 
+#[derive(Debug, thiserror::Error, Diagnostic)]
+#[error("Failed to access gene at position {0}")]
+#[diagnostic(help = "Check that the gene is long enough")]
```

Comment:
I would guess that it's more common that the index is out of range rather than the gene is "too short". What sort of error does `.get()` provide when you try to access an element out of range on something like a `Vec`?